tests/Unit/PointwiseFunctions/AnalyticSolutions/Elasticity/Test_HalfSpaceMirror.cpp

```diff
@@ -0,0 +1,159 @@
+#include "PointwiseFunctions/Elasticity/ConstitutiveRelations/IsotropicHomogeneous.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace {
+
+struct HalfSpaceMirrorProxy : Elasticity::Solutions::HalfSpaceMirror {
+  using Elasticity::Solutions::HalfSpaceMirror::HalfSpaceMirror;
+
+  using field_tags = tmpl::list<Elasticity::Tags::Displacement<3>,
+                                Elasticity::Tags::Strain<3>>;
+  using source_tags =
+      tmpl::list<Tags::FixedSource<Elasticity::Tags::Displacement<3>>>;
+
+  tuples::tagged_tuple_from_typelist<field_tags> field_variables(
+      const tnsr::I<DataVector, 3>& x) const noexcept {
+    return Elasticity::Solutions::HalfSpaceMirror::variables(x, field_tags{});
+  }
+
+  // NOLINTNEXTLINE(readability-convert-member-functions-to-static)
+  tuples::tagged_tuple_from_typelist<source_tags> source_variables(
+      const tnsr::I<DataVector, 3>& x) const noexcept {
+    return Elasticity::Solutions::HalfSpaceMirror::variables(x, source_tags{});
+  }
+};
+
+}  // namespace
+
+SPECTRE_TEST_CASE(
+    "Unit.PointwiseFunctions.AnalyticSolutions.Elasticity.HalfSpaceMirror",
+    "[PointwiseFunctions][Unit][Elasticity]") {
+  const size_t dim = Elasticity::Solutions::HalfSpaceMirror::volume_dim;
+  const Elasticity::Solutions::HalfSpaceMirror check_solution{
+      0.177, Elasticity::ConstitutiveRelations::IsotropicHomogeneous<3>{
+                 // fused silica: E=72, nu=0.17
+                 // K =\frac{E}{3(1-2\nu)}, \mu =\frac{E}{2(1+\nu)}
+                 36.36363636363637, 30.76923076923077}};
+  const auto created_solution =
+      TestHelpers::test_creation<Elasticity::Solutions::HalfSpaceMirror>(
+          "BeamWidth: 0.177\n"
+          "Material:\n"
+          "  BulkModulus: 36.36363636363637\n"
+          "  ShearModulus: 30.76923076923077\n");
+  CHECK(created_solution == check_solution);
+  test_serialization(check_solution);
+
+  pypp::SetupLocalPythonEnvironment local_python_env{"PointwiseFunctions"};
+  const Elasticity::ConstitutiveRelations::IsotropicHomogeneous<dim>
+      constitutive_relation{36.36363636363637, 30.76923076923077};
+  const HalfSpaceMirrorProxy solution{0.177, constitutive_relation, 350};
+  pypp::check_with_random_values<1,
+                                 tmpl::list<Elasticity::Tags::Displacement<dim>,
+                                            Elasticity::Tags::Strain<dim>>>(
+      &HalfSpaceMirrorProxy::field_variables, solution,
+      "AnalyticSolutions.Elasticity.HalfSpaceMirror",
+      {"displacement", "strain"}, {{{0., 3.}}},
+      std::make_tuple(0.177, 36.36363636363637, 30.76923076923077),
+      DataVector(5));
+  pypp::check_with_random_values<
+      1, tmpl::list<Tags::FixedSource<Elasticity::Tags::Displacement<dim>>>>(
+      &HalfSpaceMirrorProxy::source_variables, solution,
+      "AnalyticSolutions.Elasticity.HalfSpaceMirror", {"source"}, {{{0., 3.}}},
+      std::make_tuple(), DataVector(5));
```

Comment:
One of your random tests failed on CI. The reason is that the test generates new random data every time it runs, so it may only sometimes fail (independence of whether it runs on gcc or clang, for example). The issue is that the tolerances used for the numerical integrals in C++ and Python and the tolerance for the `CHECK`s are not consistent:
- In Python you are using an absolute tolerance of 1e-13, which seems fine.
- The default tolerances for your C++ implementation are 1e-12 abs and 1e-10 rel, which seem fine as well, but for this test you should probably choose and write them explicitly in line 81, e.g. make them 1e-12 abs and 0. rel.
- The `CHECK` tolerance is defined by `pypp::check_with_random_values`, which takes an optional argument `epsilon` that defaults to `1e-12`. That's what causes these random failures. Choose a tolerance higher than what you use for the numerical integrals, e.g. 1e-10, and pass it to `check_with_random_values`.

Reply:
I will do that, thank you!

Reply:
Also set the abs and rel tolerances explicitly in line 81